constantine/math/elliptic/ec_twistededwards_affine.nim

```diff
@@ -121,7 +122,10 @@ func trySetFromCoordX*[F](P: var ECP_TwEdwards_Aff[F], x: F): SecretBool =
   P.x += one
 
   # √((1 - ax²)/(1 - dx²))
-  result = sqrt_ratio_if_square(t, P.x, P.y)
+  var temp {.noInit.}: F
+  temp.inv(P.y)
+  temp.prod(P.x, temp)
+  result = t.sqrtPrecomp(temp) # TODO: needs to be made conditional
```

Comment:
`sqrtPrecomp` is only defined for Banderwagon / Bandersnatch and will break for Curve25519 for example.
Furthermore it is not constant-time and may handle secrets, so it needs to be explicitly suffixed with vartime as well as its callers.

Instead `invsqrt_vartime` should be created with special handling that uses the dlog for Banderwagon / Bandersnatch, similar to:
https://github.com/mratsim/constantine/blob/5894a8da757bb935b681c73d1fdc92a7838f94ed/constantine/math/arithmetic/finite_fields_square_root.nim#L198-L214
And a new `trySetFromCoordX_vartime` should be used.